pkg/scheduler/internal/queue/scheduling_queue.go

```diff
@@ -686,9 +772,45 @@ func (p *PriorityQueue) Pop() (*framework.QueuedPodInfo, error) {
 	pInfo := obj.(*framework.QueuedPodInfo)
 	pInfo.Attempts++
 	p.schedulingCycle++
+	// In flight, no move request yet.
+	p.inFlightPods[pInfo.Pod.UID] = inFlightPod{
```

Comment:
I think we should place the element of the event list (`receivedEvents`) as the value here. So that, this struct along with `receivedEvents` form a double-linked list, which will ease locating a pod (like in `DonePod()`).

Reply:
You mean this idea, right?
https://github.com/kubernetes/kubernetes/pull/118438#discussion_r1252296619

Reply:
yes ;)